### ISSUE TITLE
用tf.metrics.Mean()记录损失，进入新的epoch时损失出现跳跃性下降。

### ISSUE THREAD

**Issue by YuhangJi** [calmisential/YOLOv3_TensorFlow2#17]
我正在用tf.metrics.Mean()记录训练的损失，如果按照每一轮训练结束后重置loss metrics，那么在新的一轮训练中输出的loss值会骤降。现实训练中，第一轮训练训练loss会从几千降低到几百，然后进入下一轮时，损失直接突变为几十，请问此时的loss值是否是准确的呢？

**Comment by YuhangJi** [comment]
损失函数采用继承keras.losses.Loss，但查阅官方文档，提示最后的输出维度必须包含batch。同时我也查阅了其它实现，其损失函数返回值维度为[batch,1]。推测作者的损失函数有待改进，我对作者的代码进行了部分调整，拓展了自定义数据集的灵活性，这版本的代码仍是一个不错的学习范例！损失函数的完善希望得到作者的支持！感谢！
